### PR TITLE
pythonPackages.roku: init at v4.1

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change

cc: @Mic92 

add `pythonPacakges.roku`, regenerate home-assistant's `component-packages.nix`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by colemickens** [comment]
```
> nixpkgs-review rev HEAD
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
$ git worktree add /home/cole/.cache/nixpkgs-review/rev-3203a2c6a78ede4baaed3e36da4af3bf4a329cf3/nixpkgs 7dc2439a1be07be4ed6a0e0fb9f6e6e3c97ab126
Preparing worktree (detached HEAD 7dc2439a1be)
Updating files: 100% (21298/21298), done.
HEAD is now at 7dc2439a1be Merge pull request #83515 from snicket2100/firejail-fix
$ nix-env -f /home/cole/.cache/nixpkgs-review/rev-3203a2c6a78ede4baaed3e36da4af3bf4a329cf3/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit 3203a2c6a78ede4baaed3e36da4af3bf4a329cf3
Auto-merging pkgs/top-level/python-packages.nix
Auto-merging pkgs/servers/home-assistant/component-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix-env -f /home/cole/.cache/nixpkgs-review/rev-3203a2c6a78ede4baaed3e36da4af3bf4a329cf3/nixpkgs -qaP --xml --out-path --show-trace --meta
2 package added:
python3.7-roku (init at 4.1) python3.8-roku (init at 4.1)

39 package updated:
bcc bpftrace caffe can-isotp chipsec clang-wrapper-with-reexport-hack digimend-unstable digitalbitbox evdi-unstable evdi-unstable flit-build-hook libselinux mba6x_bl musl newlib nix-linter-unstable octave php-couchbase php-couchbase php-pcs php-pcs php-php_excel php-php_excel php-php_excel php-php_excel php-php_excel php-php_excel poezio prl-tools pyblock pyblock python-notify python-notify python2.7-habanero python2.7-handout python2.7-hass-nabucasa qes r8125 r8168

$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/cole/.cache/nixpkgs-review/rev-3203a2c6a78ede4baaed3e36da4af3bf4a329cf3/build.nix
[1 built, 0.0 MiB DL]
39 package marked as broken and skipped:
clang-sierraHack digitalbitbox linuxPackages_4_4.evdi linuxPackages_hardkernel_4_14.bcc linuxPackages_hardkernel_4_14.bpftrace linuxPackages_hardkernel_4_14.can-isotp linuxPackages_hardkernel_4_14.chipsec linuxPackages_hardkernel_4_14.digimend linuxPackages_hardkernel_4_14.evdi linuxPackages_hardkernel_4_14.mba6x_bl linuxPackages_hardkernel_4_14.prl-tools linuxPackages_hardkernel_4_14.r8125 linuxPackages_hardkernel_4_14.r8168 muslCross newlibCross nix-linter octave-jit php72Packages-unit.php_excel php72Packages.php_excel php73Packages-unit.php_excel php73Packages.php_excel php74Packages-unit.couchbase php74Packages-unit.pcs php74Packages-unit.php_excel php74Packages.couchbase php74Packages.pcs php74Packages.php_excel python27Packages.caffe python27Packages.flitBuildHook python27Packages.habanero python27Packages.handout python27Packages.hass-nabucasa python37Packages.notify python37Packages.pyblock python38Packages.libselinux python38Packages.notify python38Packages.poezio python38Packages.pyblock qes

2 package built:
python37Packages.roku python38Packages.roku

```

File: pkgs/development/python-modules/roku/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "jcarbaugh";
+    repo = "python-roku";
+    rev = "v${version}";
+    sha256 = "09mq59kjll7gj1srw4qc921ncsm7cld95sbz5v3p2bwmgckpqza7";
+  };
+
+  buildInputs = [ requests ];
+
+  doCheck = true;
```
Comment (jonringer) [inline pkgs/development/python-modules/roku/default.nix]: default with buildPythonPAckage
```suggestion
```

File: pkgs/development/python-modules/roku/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "jcarbaugh";
+    repo = "python-roku";
+    rev = "v${version}";
+    sha256 = "09mq59kjll7gj1srw4qc921ncsm7cld95sbz5v3p2bwmgckpqza7";
+  };
+
+  buildInputs = [ requests ];
```
Comment (jonringer) [inline pkgs/development/python-modules/roku/default.nix]:
```suggestion
  propagatedBuildInputs = [ requests ];
```

File: pkgs/servers/home-assistant/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+.mypy_cache/
```
Comment (jonringer) [inline pkgs/servers/home-assistant/.gitignore]: I would put this in `.git/info/exclude`
```suggestion
```

**Review by jonringer — COMMENTED** [review]
you can leave off the `v` in your commit messages and PR titles, it's understood that it denotes a version. Where as in git, it's not obvious that branch `4.1` means a version number

File: pkgs/servers/home-assistant/.gitignore (hunk 3 above)
Comment (Mic92) [reply pkgs/servers/home-assistant/.gitignore]: We have some mypy stuff now in nixpkgs. Maybe we can also add it to the global .gitignore.

**Pull request merged**